crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -2812,14 +2812,168 @@ impl Analyzer<'_, '_> {
 
     /// Should be called only if `to` is not expandable.
     fn assign_to_intrinsic(&mut self, data: &mut AssignData, to: &StringMapping, r: &Type, opts: AssignOpts) -> VResult<()> {
-        match to.kind {
-            IntrinsicKind::Uppercase => {}
-            IntrinsicKind::Lowercase => {}
-            IntrinsicKind::Capitalize => {}
-            IntrinsicKind::Uncapitalize => {}
+        match r {
+            Type::Lit(LitType {
+                lit: RTsLit::Str(value), ..
+            }) => match to.kind {
+                IntrinsicKind::Uppercase => {
+                    if let Some(value) = &value.raw {
+                        if !value.to_uppercase().eq(&value.to_string()) {
+                            return Err(ErrorKind::AssignFailed {
+                                span: r.span(),
+                                left: box Type::StringMapping(to.clone()),
+                                right_ident: None,
+                                right: box r.clone(),
+                                cause: vec![],
+                            }
+                            .into());
+                        }
+                    }
+                }
+                IntrinsicKind::Lowercase => {
+                    if let Some(value) = &value.raw {
+                        if !value.to_lowercase().eq(&value.to_string()) {
```

Comment:
Same here. `JdWord` implements `Deref<Target = str>` so you can do `&**value` to get `&str`

Reply:
done,  clippy made me do

```diff
- &**value
+ **value
```
though, I hope that's ok